python_modules/dagster/dagster/_core/storage/partition_status_cache.py

```diff
@@ -397,18 +425,19 @@ def build_failed_and_in_progress_partition_subset(
 def get_and_update_asset_status_cache_value(
     instance: DagsterInstance,
     asset_key: AssetKey,
-    partitions_def: Optional[PartitionsDefinition] = None,
+    partitions_def: Optional[PartitionsDefinition],
     dynamic_partitions_loader: Optional[DynamicPartitionsStore] = None,
-    asset_record: Optional["AssetRecord"] = None,
+    batch_asset_record_loader: Optional[BatchAssetRecordLoader] = None,
 ) -> Optional[AssetStatusCacheValue]:
-    asset_record = asset_record or next(
-        iter(instance.get_asset_records(asset_keys=[asset_key])), None
-    )
```

Comment:
we now pass in a BatchAssetRecordLoader here, and now "asset_record=None" means "there is no asset record", not "you should go check whether or not there is an asset record", which makes things faster